### ISSUE TITLE
Errors not loaded into model's errors attribute

### ISSUE THREAD

**Issue by dmarkow** [emberjs/data#3371]
When I submit an invalid request to my API, it returns some JSON with a 422 status.
```js
var user = this.store.createRecord("user");
user.save(); // This triggers a 422 error
```

The JSON returned is along the lines of:

```json
{"errors":{"login":["can't be blank"]}}
```

In 1.0.0.beta-19.1 and earlier, this would allow me to use `user.get("errors")` to iterate over each error. As of 1.13.0, however, after the 422 response is received, `user.errors` is still empty.

I've been using Ember Data's built-in Active Model Adapter/Serializer; I also tried using the ember-data/active-model-adapter addon but ran into the same problem.

Has the approach for accessing a model's errors changed as of 1.13.0?

**Comment by igorT** [comment]
It has. , but you should still be getting a useful deprecation.
Please look under https://gist.github.com/tchak/3d28e8af06d7101127d1
 cc @tchak

**Comment by dmarkow** [comment]
`user.get("error.errors")` returns undefined.

If it helps, here are the specific console results for each version.

1.0.0-beta.19.1:

```
DEPRECATION: Ember.reduceComputed is deprecated. Replace it with plain array methods
        at Descriptor.ReduceComputedProperty (http://localhost:4200/assets/vendor.js:40073:31)
        at Object.reduceComputed (http://localhost:4200/assets/vendor.js:40459:14)
        at http://localhost:4200/assets/vendor.js:75870:36
        at http://localhost:4200/assets/vendor.js:81831:4
DEBUG: -------------------------------
DEBUG: Ember      : 1.13.0
DEBUG: Ember Data : 1.0.0-beta.19.1
DEBUG: jQuery     : 1.11.3
DEBUG: -------------------------------
Ember Inspector Active
Ember Inspector ($E):  Class {_qpDelegate: undefined, currentPath: "index", currentRouteName: "index", __ember1434577720046: "ember425", __nextSuper: undefined…}
user = $E.store.createRecord("user")
Class {id: null, store: Class, container: Container, _internalModel: InternalModel, currentState: Object…}
user.save()
Class {__ember1434577720046: null, __nextSuper: undefined, __ember_meta__: Object}
POST http://localhost:4200/api/v1/users 422 (Unprocessable Entity)
Error: The backend rejected the commit because it was invalid: {password: can't be blank, email: has already been taken,can't be blank,has already been taken, billing_position: can't be blank, login: can't be blank, start_date: can't be blank}
    at new Error (native)
    at Error.EmberError (http://localhost:4200/assets/vendor.js:25627:21)
    at Error.ember$data$lib$system$model$errors$invalid$$InvalidError (http://localhost:4200/assets/vendor.js:68369:62)
    at http://localhost:4200/assets/vendor.js:79763:20
    at tryCatch (http://localhost:4200/assets/vendor.js:65249:14)
    at invokeCallback (http://localhost:4200/assets/vendor.js:65264:15)
    at publish (http://localhost:4200/assets/vendor.js:65232:9)
    at publishRejection (http://localhost:4200/assets/vendor.js:65166:5)
    at http://localhost:4200/assets/vendor.js:42056:7
    at Queue.invoke (http://localhost:4200/assets/vendor.js:11455:16)
user.get("error")
undefined
user.get("errors")
Class {__ember1434577720046: "ember500", __nextSuper: undefined, __ember_meta__: Object}__ember1434577720046: "ember500"__ember_meta__: Object__nextSuper: undefinedcontent: (...)get content: GETTER_FUNCTION() {set content: SETTER_FUNCTION(value) {length: (...)get length: GETTER_FUNCTION() {set length: SETTER_FUNCTION(value) {__proto__: Class
user.get("error.errors")
undefined
user.get("errors.length")
7
```

1.13.0:

```
DEBUG: -------------------------------
DEBUG: Ember      : 1.13.0
DEBUG: Ember Data : 1.13.0
DEBUG: jQuery     : 1.11.3
DEBUG: -------------------------------
Ember Inspector Active
Ember Inspector ($E):  Class {_qpDelegate: undefined, currentPath: "index", currentRouteName: "index", __ember1434577623728: "ember435", __nextSuper: undefined…}
user = $E.store.createRecord("user")
Class {id: null, store: Class, container: Container, _internalModel: InternalModel, currentState: Object…}
user.save()
Class {__ember1434577623728: null, __nextSuper: undefined, __ember_meta__: Object}
POST http://localhost:4200/api/v1/users 422 (Unprocessable Entity)
Error: Unprocessable Entity
    at ember$data$lib$system$adapter$$default.extend.ajax.Ember.RSVP.Promise.hash.error (rest-adapter.js:802)
    at jQuery.Callbacks.fire (jquery.js:3148)
    at Object.jQuery.Callbacks.self.fireWith [as rejectWith] (jquery.js:3260)
    at done (jquery.js:9316)
    at XMLHttpRequest.jQuery.ajaxTransport.send.callback (jquery.js:9718)
user.get("error")
Error: Unprocessable Entity
    at ember$data$lib$system$adapter$$default.extend.ajax.Ember.RSVP.Promise.hash.error (http://localhost:4200/assets/vendor.js:70334:29)
    at jQuery.Callbacks.fire (http://localhost:4200/assets/vendor.js:3350:30)
    at Object.jQuery.Callbacks.self.fireWith [as rejectWith] (http://localhost:4200/assets/vendor.js:3462:7)
    at done (http://localhost:4200/assets/vendor.js:9518:14)
    at XMLHttpRequest.jQuery.ajaxTransport.send.callback (http://localhost:4200/assets/vendor.js:9920:8)
user.get("errors")
Class {__ember1434577623728: null, __nextSuper: undefined, __ember_meta__: Object}
user.get("error.errors")
undefined
```

**Comment by tchak** [comment]
I can confirm. Error handler is broken :( So sorry about that. Fix is on the way #3374.

**Comment by dmarkow** [comment]
No worries, I'm super grateful for the hard work you're all doing. Thanks!
